connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -270,6 +273,8 @@ public boolean startConnector(
             if (existing != null)
                 throw new ConnectException("Connector with name " + connName + " already exists");
 
+            executor.submit(workerConnector);
```

Comment:
As far as the API provided by the `Worker` class goes, yes, it is possible and there are no guards in place to prevent that from happening. However, at the moment both herders are well-behaved in that they do not instruct the `Worker` to start tasks until after the connector has finished starting as well. This is pretty simple to verify, and unlikely to change, since the herders wait for connectors to start before requesting task configurations from them, which in turn are required in order to actually go about starting tasks.